pruner.h

```diff
@@ -9,16 +9,26 @@ namespace P4PRUNER {
 
 class Pruner : public Transform {
   public:
-    // IR::P4Program *program;
-    Pruner() {
+    std::vector<const IR::Statement *> to_prune;
+    Pruner(std::vector<const IR::Statement *> _to_prune) {
```

Comment:
Instead of a vector of `IR::Statement *` this should be a vector of `IR::Statement` since that is what we are storing and comparing. You will have to change signatures everywhere

Reply:
I tried doing that, but if i 
1. make a vector of `const IR::Statement`, it complains saying that the elements must be fluid, non-constant
2. make a vector of `IR::Statement`, it complains saying this
```
/home/roborobo/projects/gauntlet_paper/gauntlet/modules/p4c/extensions/pruner/pruner.cpp:26:30:`   required from here
/usr/include/c++/9/ext/new_allocator.h:145:20: error: invalid new-expression of abstract class type ‘IR::Statement’
  145 |  noexcept(noexcept(::new((void *)__p)
      |                    ^~~~~~~~~~~~~~~~~~
  146 |        _Up(std::forward<_Args>(__args)...)))
```

Reply:
Okay let's scratch that for now.